### PR TITLE
Website: Generate API documentation when building site.

### LINKED ISSUE #183: Generate JS API documentation for the website
The source for JS API documentation is `src/api.js`.  We should generate documentation for the API on the website from that file.

### PR DESCRIPTION
**Pull request by nicksay**
Closes #183.

### CODE REVIEW

File: web/Rakefile [hunk 1]
```diff
@@ -79,7 +87,7 @@ all_to_copy.each do |dst, src|
   file dst => src do
     dstdir = File.dirname(dst)
     FileUtils.mkdir_p(dstdir) unless File.directory?(dstdir)
-    puts("cp #{src} #{dst}") if verbose
+    puts("cp #{src} #{dst}") if verbose == true
```
Comment (awbraunstein) [inline web/Rakefile]: Does ruby not like `if verbose`?

Comment (nicksay) [inline web/Rakefile]: Apparently Rake sets the value to the `:default` symbol so you need to explicitly check for `verbose == true`:
http://engineering.hoteltonight.com/checking-if-rake-was-called-with-verbose-or-t

Comment (awbraunstein) [inline web/Rakefile]: weird.

**Comment by awbraunstein** [comment]
LGTM

**Pull request merged**